src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -0,0 +1,542 @@
+        if (!(entity instanceof TileEntityFurnace)
+                || ((TileEntityFurnace) entity).isBurning()
+                || (ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(2))
+                && ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(0))))
+        {
+            walkTo = null;
+            return START_WORKING;
+        }
+        walkTo = null;
+
+        InventoryUtils.transferItemStackIntoNextFreeSlotInItemHandlers(
+                new InvWrapper((TileEntityFurnace) entity), 2,
+                new InvWrapper(worker.getInventoryCitizen()));
+
+        if (!ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(0)))
+        {
+            if (!InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel))
+            {
+                walkTo = null;
+                return COOK_GET_FIREWOOD;
+            }
+
+            InventoryUtils.transferItemStackIntoNextFreeSlotInItemHandlers(
+                    new InvWrapper(worker.getInventoryCitizen()),
+                    InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel),
+                    new InvWrapper((TileEntityFurnace) entity));
+        }
+
+        incrementActionsDone();
+        setDelay(STANDARD_DELAY);
+        return START_WORKING;
+    }
+
+    private AIState gatherFoodFromBuilding()
+    {
+        if(needsCurrently == null)
+        {
+            needsCurrently = isFood;
+        }
+
+        if(InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), needsCurrently))
+        {
+            return COOK_SERVE;
+        }
+
+        final BlockPos pos = getOwnBuilding().getTileEntity().getPositionOfChestWithItemStack(needsCurrently);
+        if (pos == null)
+        {
+            return START_WORKING;
+        }
+
+        if (walkToBlock(pos))
+        {
+            return getState();
+        }
+
+        final boolean transfered = tryTransferFromPosToCook(pos, needsCurrently);
+        if (transfered)
+        {
+            return COOK_SERVE;
+        }
+        //todo
+        setDelay(STANDARD_DELAY);
+        return START_WORKING;
+    }
+
+    private AIState serveFood()
+    {
+        if (citizenToServe.isEmpty())
+        {
+            return START_WORKING;
+        }
+
+        if (walkToBlock(citizenToServe.get(0).getPosition()))
+        {
+            setDelay(2);
+            return getState();
+        }
+
+        InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                new InvWrapper(worker.getInventoryCitizen()),
+                isFood,
+                AMOUNT_OF_FOOD_TO_SERVE,
+                new InvWrapper(citizenToServe.get(0).getInventoryCitizen()));
+
+        citizenToServe.remove(0);
+        setDelay(SERVE_DELAY);
+        return getState();
+    }
+
+    private AIState cookFood()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)